### PR TITLE
Adds development standards

### PR DESCRIPTION
**Pull request by DannyGB**
Initial stab at writing up development standards currently in-use

### CODE REVIEW

**Review by hardik-desai-kcl — APPROVED** [review]

File: collections/_development_standards/patterns.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Below are described the major engineering patterns currently in-use across the Service. There is no definitive list as it is understood that patterns should be used when and where appropriate.
+
+### Orchestrator
+
+As controllers age they are asked to do more and more work which can result in serious bloat. We want to keep our controllers focused on what they should be doing such as handling requests, invoking business rules and returning responses. So to keep them light we need to hand-off any other work that may end up in a controller out to somewhere else and the Orchestrator can be used to fill this need. 
+
+The use of Orchestrators can also make a codebase more testable by moving logic and flow out of controllers and their tight coupling to the runtime.
+
+![Orchestrator Overview](./images/OrchestratorOverview.png)
```
Comment (dashton82) [inline collections/_development_standards/patterns.md]: We have moved away from this pattern over time of creating separate orchestrator classes - the controller actions are normally very light weight anyway and only tend to have calls to mediatr and then building a view model from it. All tests are isolated that way and the tests on the controller are normally very light weight, letting the controller do its intended orchestrator role.

**Review by dashton82 — COMMENTED** [review]
All looks good - only one observation 👍

File: collections/_development_standards/patterns.md (hunk 1 above)
Comment (DannyGB) [reply collections/_development_standards/patterns.md]: That makes sense to me. I'll leave in the Orchestrator section for historical purposes and add comments around the reasoning behind move away from the pattern.

I do have some reservations about Mediatr though as I think it makes the code harder to read

**Review by dashton82 — APPROVED** [review]

File: collections/_development_standards/patterns.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+#### References
+
+* [Simplify Your Architecture Using the Orchestrator Pattern](http://www.jamiemaguire.net/index.php/2017/05/06/simplify-your-architecture-using-the-orchestrator-pattern/)
+* [The Orchestrator Pattern](http://www.michaeltaylorp3.net/orchestrator-pattern/)
+* [Never Mind the Controller, Here is the Orchestrator](https://www.simple-talk.com/dotnet/asp-net/never-mind-the-controller-here-is-the-orchestrator/)
+  
+### Mediator
+
+With the mediator pattern, communication between objects is encapsulated within a mediator object and do not communicate directly with each other, but instead communicate through the mediator. This reduces the dependencies between communicating objects and reduces coupling.
```
Comment (MichaelYoung1981) [inline collections/_development_standards/patterns.md:67]: Can we add a comment that the Mediator pattern should only be used to decouple the different layers of the application as opposed to using it to reuse code within the same layer?

Comment (DannyGB) [reply collections/_development_standards/patterns.md:67]: How does this sound:

`When seeking to use the Mediatr pattern you should look to use it as a mechanism to decouple the different application layers and resist the urge to use it as a code-reuse pattern for which other approaches are more appropriate.`

**Review by satya-sanikommu — APPROVED** [review]

**Pull request merged**